### PR TITLE
Fix load pretrain weight issue in ResNet

### LINKED ISSUE #7923: Pretrained weights incompatibility for resnet10
**Describe the bug**
When creating a ResNet10 model with pretrained weights, the construction fails with
```bash
RuntimeError: Error(s) in loading state_dict for ResNet:
Missing key(s) in state_dict: "layer2.0.downsample.0.bias", "layer3.0.downsample.0.bias", "layer4.0.downsample.0.bias". 
```

**To Reproduce**

```
from monai.networks.nets import resnet10
# using monai 1.3.2

model = resnet10(pretrained=True, n_input_channels=1, feed_forward=False, spatial_dims=3)
# RuntimeError occurs
```

**Expected behavior**
The model should be created successfully with the pretrained weights from Med3D

**Environment**

Ensuring you use the relevant python executable, please paste the output of:

```
python -c "import monai; monai.config.print_debug_info()"
```

```bash
================================
Printing MONAI config...
================================
MONAI version: 1.3.2
Numpy version: 2.0.0
Pytorch version: 2.3.1+cu121
MONAI flags: HAS_EXT = False, USE_COMPILED = False, USE_META_DICT = False
MONAI rev id: 59a7211070538586369afd4a01eca0a7fe2e742e
MONAI __file__: /home/<username>/miniconda3/envs/monai/lib/python3.10/site-packages/monai/__init__.py

Optional dependencies:
Pytorch Ignite version: NOT INSTALLED or UNKNOWN VERSION.
ITK version: NOT INSTALLED or UNKNOWN VERSION.
Nibabel version: NOT INSTALLED or UNKNOWN VERSION.
scikit-image version: NOT INSTALLED or UNKNOWN VERSION.
scipy version: NOT INSTALLED or UNKNOWN VERSION.
Pillow version: NOT INSTALLED or UNKNOWN VERSION.
Tensorboard version: NOT INSTALLED or UNKNOWN VERSION.
gdown version: NOT INSTALLED or UNKNOWN VERSION.
TorchVision version: NOT INSTALLED or UNKNOWN VERSION.
tqdm version: 4.66.4
lmdb version: NOT INSTALLED or UNKNOWN VERSION.
psutil version: NOT INSTALLED or UNKNOWN VERSION.
pandas version: NOT INSTALLED or UNKNOWN VERSION.
einops version: NOT INSTALLED or UNKNOWN VERSION.
transformers version: NOT INSTALLED or UNKNOWN VERSION.
mlflow version: NOT INSTALLED or UNKNOWN VERSION.
pynrrd version: NOT INSTALLED or UNKNOWN VERSION.
clearml version: NOT INSTALLED or UNKNOWN VERSION.

For details about installing the optional dependencies, please visit:
    https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies


================================
Printing system config...
================================
`psutil` required for `print_system_info`

================================
Printing GPU config...
================================
Num GPUs: 0
Has CUDA: False
cuDNN enabled: True
NVIDIA_TF32_OVERRIDE: None
TORCH_ALLOW_TF32_CUBLAS_OVERRIDE: None
cuDNN version: 8902
```

### PR DESCRIPTION
**Pull request by KumoLiu**
Fixes #7923

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by ericspod** [comment]
The `bias_downsample` value should be bool everywhere from what I gather, however in `get_medicalnet_pretrained_resnet_args` there's still code treating it like an int, should this be revised as well?

**Comment by KumoLiu** [comment]
> The `bias_downsample` value should be bool everywhere from what I gather, however in `get_medicalnet_pretrained_resnet_args` there's still code treating it like an int, should this be revised as well?

Yes, I revised all of them in the latest commit. Could you please help review it again? Thanks!

File: monai/networks/nets/resnet.py [hunk 1]
```diff
@@ -681,7 +680,7 @@ def get_medicalnet_pretrained_resnet_args(resnet_depth: int):
     # After testing
     # False: 10, 50, 101, 152, 200
     # Any: 18, 34
-    bias_downsample = -1 if resnet_depth in [18, 34] else 0  # 18, 10, 34
+    bias_downsample = True if resnet_depth in [18, 34] else False  # 18, 10, 34
```
Comment (ericspod) [inline monai/networks/nets/resnet.py:683]:
```suggestion
    bias_downsample = resnet_depth in [18, 34]
```

Slightly simpler, I don't think the comment was accurate anyway given the values being checked.

**Review by ericspod — APPROVED** [review]

**Pull request merged**